### ISSUE TITLE
how can I obtain a double-precision genotype matrix?

### ISSUE THREAD

**Issue by ttian627** [stephenslab/susieR#149]
My genotype matrix contain "-1","0","2"，how can I convert it to a double-precision genotype matrix?

**Comment by pcarbo** [comment]
You can do 

```r
storage.mode(X) <- "double"
```

**Comment by gaow** [comment]
@ttian627 yes `storage.mode` is the way to go. But are you sure the genotype matrix is -1/0/2 not 1/0/2 ? It should be the latter.

**Comment by ttian627** [comment]
The species I study is maize. my genotype matrix is -1/0/2, for example, -1 (NN), 0 (AA), 2 (TT).
Is it correct to comput the correction matrix after convert it to double-precision? I find the example data at vignettes were the following format.
![图片](https://user-images.githubusercontent.com/47581845/146290001-1f407328-02d1-4f65-acf9-b3965ba41e5b.png)

**Comment by pcarbo** [comment]
What is "NN"? Does that mean the genotype is missing?

**Comment by ttian627** [comment]
yes, NN means missing.













At 2021-12-16 09:52:09, "Peter Carbonetto" ***@***.***> wrote:

What is "NN"? Does that mean the genotype is missing?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by pcarbo** [comment]
In R missing values are represented by "NA"; see `help("NA")`. You will need to impute the missing genotypes beforehand, or remove the SNPs with missing genotypes.

**Comment by ttian627** [comment]
> In R missing values are represented by "NA"; see `help("NA")`. You will need to impute the missing genotypes beforehand, or remove the SNPs with missing genotypes.

When we comput the LD matrix, the NA at each SNP pairs will be removed, I find the "cor" function at R can do this way by setting "na.rm=pairwise.complete.obs". Is this OK? By the way, the example genotype matrix data at vignettes, after remove redundant values at each column, their are 2-4 values for each SNP, Is this corresponding to genotypes like this, for example, (AA/TT), (AA/AT/TT) and (AA/AT/TT/NA). I really want to know how to transform the genotype matrix to this format.

**Comment by stephens999** [comment]
safer to impute the missing genotypes. results may not be reliable
otherwise.

**Comment by gaow** [comment]
> "na.rm=pairwise.complete.obs".

this may result in LD matrices that are not positive semidefinite, and is indeed not reliable.

>  I find the example data at vignettes were the following format.

The example data is column scaled version of diploid genotype data originally coded as 0/1/2 -- that is, the original genotype data is a matrix of 0/1/2 for AA, AT, TT genotypes per individual. I believe you can stick to this additive coding for your application but you'll have to handle missing data properly not using -1 in place of missing data.
